### ISSUE TITLE
vue/valid-v-on errors on working code

### ISSUE THREAD

**Issue by KaelWD** [vuejs/eslint-plugin-vue#2251]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** 8.43.0
- **eslint-plugin-vue version:** 9.15.0
- **Node version:** 18.16.0
- **Operating System:** Linux

**Please show your full configuration:**

```json5
extends: ["plugin:vue/vue3-essential"],
rules: {
    "vue/multi-word-component-names": "off",
    "vue/no-dupe-keys": "off",
    "vue/no-unused-components": "off",
    "vue/no-v-model-argument": "off",
    "vue/valid-v-model": "off",
    "vue/valid-v-slot": ["error", {
        "allowModifiers": true
    }],
}
```

**What did you do?**	

```vue
<v-text-field
        ref="textFieldRef"
        :model-value="masked"
        @update:model-value=""
/>
```

**What did you expect to happen?**
Nothing

**What actually happened?**

```
5:9  error  'v-on' directives require a value or verb modifier (like 'stop' or 'prevent')  vue/valid-v-on
```

**Repository to reproduce this issue**

This is simple enough I don't think it needs one. `@update:model-value` should be an error because vue doesn't like it either, but `@update:model-value=""` actually works fine and just throws an eslint warning for no reason.

**Comment by FloEdelmann** [comment]
Why do you want to use `@update:model-value=""`? It doesn't do anything, does it?

**Comment by KaelWD** [comment]
It does whatever the component wants it to do, in this case it stops the input from keeping its own internal state. This is pretty common with other components too, like `<v-list-item @click="" />` adds a hover effect.

**Comment by FloEdelmann** [comment]
I'm not sure whether it should be allowed as a default. But I can see your use case. What do you think about an `allowEmpty: true` parameter (false by default)?

**Comment by KaelWD** [comment]
I don't really like how many configuration flags this plugin has added over time, most of which have somewhat puzzling defaults. If the intention of this rule is to prevent errors then empty events should be allowed, because they either do nothing or are intentional. If the intention is to prevent mistakes then it should still be allowed by default because sometimes it isn't a mistake.
